Laundromat/src/main/java/org/ds/model/machine/WashingMachine.java

```diff
@@ -0,0 +1,72 @@
+package org.ds.model.machine;
+
+import jakarta.persistence.*;
+
+import java.util.Date;
+
+/**
+ * Класс Стиральная машинка - хранит в себе информацию о конкретной машинке.
+ * Хранится в базе данных таблицей machines.
+ * Сущность зависит от общаги.
+ */
+@Entity
+@Table(name = "machines")
+public class WashingMachine {
+    @Id
+    @GeneratedValue(strategy = GenerationType.SEQUENCE)
+    private Long id;
+    @Column(name = "state")
+    private State state;
+    @Column(name = "last_time")
+    private Long time;
+    @Column(name = "duration")
+    private Long duration;
+
+    public WashingMachine() {
+
+    }
+
+    public WashingMachine(Long id, State state, Long time, Long duration) {
+        this.id = id;
+        this.state = state;
+        this.time = time;
+        this.duration = duration;
+    }
+
+    public Long getId() {
+        return id;
+    }
+
+    public void setId(Long id) {
+        this.id = id;
+    }
+
+    public State getState() {
+        return state;
+    }
+
+    public void setState(State state) {
+        this.state = state;
+    }
+
+    public Long getTime() {
+        return time;
+    }
+
+    public void setTime(Long time) {
+        this.time = time;
+    }
+
+    public Long getDuration() {
+        return duration;
+    }
+
+    public void setDuration(Long duration) {
+        this.duration = duration;
+    }
+
+    public boolean isFinish() {
+        return new Date().getTime() -
+                time >= duration * 1000 || state == State.FREE;
```

Comment:
1. Магическое число
2. Негуманное отношение к памяти. Не обязательно создавать новый объект, можно использовать System.currentTimeMillis()
3. Негуманное отношение к производительности. Вы вообще знаете что такое оператор `||`? Да и форматирование некрасивое, логическое выражение разорвали, а второе склеили с обрывком первого.
```suggestion
        return state == State.FREE
               || System.currentTimeMillis() - time >= duration * 1000;
```

Reply:
А что правда duration для каждой машинки свой? Как-то это странно

Reply:
duration действительно разный для каждой машинки. Это зависит от типа стирки, одекжды и т.д.
Время стирки может занять как 20 минут, так и час. Время выбирает пользователь )

Reply:
Форматирование поправил. Не совсем понял оператор ||. Знаю что лучше указать сначала проверку на состояние, а потом уже на время, т.к. если первое верно, то дальшше он не будет проверять условие и вернет true.

Reply:
Да, вы всё правильно поняли. Простые операции лучше делать вначале, ведь если простое сравнение будет истинно, то до сложного уже дело не дойдёт.

А магические числа лучше выносить в константу, а не переменную.